internal/connector/connector.go

```diff
@@ -30,12 +31,17 @@ import (
 )
 
 type Options struct {
-	Server        string
-	Name          string
-	AccessKey     string
-	CACert        string
-	CAKey         string
-	SkipTLSVerify bool
+	Server ServerOptions
+	Name   string
+	CACert string
+	CAKey  string
+}
+
+type ServerOptions struct {
+	URL                string
+	AccessKey          string
+	SkipTLSVerify      bool
+	TrustedCertificate types.StringOrFile
```

Comment:
I think adding more TLS options to the top level config is not great. There are two separate trust roots here, and it needs to be clear which settings apply to each.

Unfortunately creating this extra structure in the config causes a few problems. It changes the name of the flags. We can make that backwards compatible if we need to.

It also makes the helm configmap for the connector more difficult because `accessKey` can come from different places.

Reply:
What does this the upgrade process look like for this new config? Could this cause unexpected behavior?

Reply:
Good question! If someone is running the connector directly they would have to change how they configure it. I believe right now we are only supporting running the connector via the helm chart. In the helm chart we are using the old values and translating it into the new structure, so anyone using the helm chart won't notice the change in structure.

I think we should change the structure in the helm chart eventually as well. I tried it initially but it got pretty involved, so I thought we could wait a bit to try and bundle it with other changes.

Reply:
Thanks, that doesnt sound that concerning

Reply:
@mxyng does that sound right to you as well?

I guess an old configmap will not be compatible with a new connector, and vice versa, but any `helm upgrade` should change both together.

Reply:
The `configmap` will be updated automatically but the service isn't configured for dynamic reloads. However, Helm upgrades will roll out new pods if the config changes. If the pod image version is updated at the same time, everything will work as expected since the config and service are upgraded in sync. If, for whatever reason, the new pod starts with the old service version, e.g. if the user pinned the connector image tag, the new pod will error